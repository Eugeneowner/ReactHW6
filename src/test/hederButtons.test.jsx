import Buttons from "../componets/header/buttons/buttons";
import {render, screen} from '@testing-library/react'
import { BrowserRouter } from "react-router-dom";
describe("Header buttons component", () => {
    it("Check render button basket", () => {
        const countBuy = 20
        render(
            <BrowserRouter>
             <Buttons countBuy={countBuy}/>
            </BrowserRouter>
            
            )
        expect(screen.getByText(countBuy)).toBeInTheDocument();

    })
    it("Create snapshot button basket", () => {
        const countBuy = 20
        const btn = render(
            <BrowserRouter>
             <Buttons countBuy={countBuy}/>
            </BrowserRouter>
        )
            expect(btn).toMatchSnapshot()
    })
    it("Check render button countWish", () => {
        const countWish = 30 
        render(
            <BrowserRouter>
                <Buttons countWish={countWish}/>
            </BrowserRouter>
        )
        expect(screen.getByText(countWish)).toBeInTheDocument();
    })
    it("Create snapshot count wish", () => {
        const countWish = 30
        const btnWish = render(
            <BrowserRouter>
                <Buttons countWish={countWish}/>
            </BrowserRouter>
        )
        expect(btnWish).toMatchSnapshot();
    })
    })
