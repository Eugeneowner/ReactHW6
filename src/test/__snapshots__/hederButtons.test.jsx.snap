// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header buttons component Create snapshot button basket 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="header_buttons"
      >
        <a
          class=""
          href="/basket"
        >
          <button
            class="header_buttons_next"
          >
            <div
              class="header_buttons_next_count"
            >
              <p
                class="header_buttons_next_count_text"
              >
                20
              </p>
            </div>
            <img
              alt="basketIco"
              class="header_buttons_next_img"
              src="basket.png"
            />
          </button>
        </a>
        <a
          class=""
          href="/wishlist"
        >
          <button
            class="header_buttons_next2"
          >
            <div
              class="header_buttons_next_count"
            >
              <p
                class="header_buttons_next_count_text"
              />
            </div>
            <img
              alt="fovouriteIcon"
              class="header_buttons_next2_img2"
              src="favorite.png"
            />
          </button>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="header_buttons"
    >
      <a
        class=""
        href="/basket"
      >
        <button
          class="header_buttons_next"
        >
          <div
            class="header_buttons_next_count"
          >
            <p
              class="header_buttons_next_count_text"
            >
              20
            </p>
          </div>
          <img
            alt="basketIco"
            class="header_buttons_next_img"
            src="basket.png"
          />
        </button>
      </a>
      <a
        class=""
        href="/wishlist"
      >
        <button
          class="header_buttons_next2"
        >
          <div
            class="header_buttons_next_count"
          >
            <p
              class="header_buttons_next_count_text"
            />
          </div>
          <img
            alt="fovouriteIcon"
            class="header_buttons_next2_img2"
            src="favorite.png"
          />
        </button>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header buttons component Create snapshot count wish 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="header_buttons"
      >
        <a
          class=""
          href="/basket"
        >
          <button
            class="header_buttons_next"
          >
            <div
              class="header_buttons_next_count"
            >
              <p
                class="header_buttons_next_count_text"
              />
            </div>
            <img
              alt="basketIco"
              class="header_buttons_next_img"
              src="basket.png"
            />
          </button>
        </a>
        <a
          class=""
          href="/wishlist"
        >
          <button
            class="header_buttons_next2"
          >
            <div
              class="header_buttons_next_count"
            >
              <p
                class="header_buttons_next_count_text"
              >
                30
              </p>
            </div>
            <img
              alt="fovouriteIcon"
              class="header_buttons_next2_img2"
              src="favorite.png"
            />
          </button>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="header_buttons"
    >
      <a
        class=""
        href="/basket"
      >
        <button
          class="header_buttons_next"
        >
          <div
            class="header_buttons_next_count"
          >
            <p
              class="header_buttons_next_count_text"
            />
          </div>
          <img
            alt="basketIco"
            class="header_buttons_next_img"
            src="basket.png"
          />
        </button>
      </a>
      <a
        class=""
        href="/wishlist"
      >
        <button
          class="header_buttons_next2"
        >
          <div
            class="header_buttons_next_count"
          >
            <p
              class="header_buttons_next_count_text"
            >
              30
            </p>
          </div>
          <img
            alt="fovouriteIcon"
            class="header_buttons_next2_img2"
            src="favorite.png"
          />
        </button>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
